test/e2e/dra/deploy.go

```diff
@@ -77,69 +77,81 @@ type Nodes struct {
 var pluginPermissions string
 
 // NewNodes selects nodes to run the test on.
+//
+// Call this outside of ginkgo.It, then use the instance inside ginkgo.It.
 func NewNodes(f *framework.Framework, minNodes, maxNodes int) *Nodes {
 	nodes := &Nodes{}
 	ginkgo.BeforeEach(func(ctx context.Context) {
-
-		ginkgo.By("selecting nodes")
-		// The kubelet plugin is harder. We deploy the builtin manifest
-		// after patching in the driver name and all nodes on which we
-		// want the plugin to run.
-		//
-		// Only a subset of the nodes are picked to avoid causing
-		// unnecessary load on a big cluster.
-		nodeList, err := e2enode.GetBoundedReadySchedulableNodes(ctx, f.ClientSet, maxNodes)
-		framework.ExpectNoError(err, "get nodes")
-		numNodes := int32(len(nodeList.Items))
-		if int(numNodes) < minNodes {
-			e2eskipper.Skipf("%d ready nodes required, only have %d", minNodes, numNodes)
-		}
-		nodes.NodeNames = nil
-		for _, node := range nodeList.Items {
-			nodes.NodeNames = append(nodes.NodeNames, node.Name)
-		}
-		sort.Strings(nodes.NodeNames)
-		framework.Logf("testing on nodes %v", nodes.NodeNames)
-
-		// Watch claims in the namespace. This is useful for monitoring a test
-		// and enables additional sanity checks.
-		claimInformer := resourceapiinformer.NewResourceClaimInformer(f.ClientSet, f.Namespace.Name, 100*time.Hour /* resync */, nil)
-		cancelCtx, cancel := context.WithCancelCause(context.Background())
-		var wg sync.WaitGroup
-		ginkgo.DeferCleanup(func() {
-			cancel(errors.New("test has completed"))
-			wg.Wait()
-		})
-		_, err = claimInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
-			AddFunc: func(obj any) {
-				defer ginkgo.GinkgoRecover()
-				claim := obj.(*resourceapi.ResourceClaim)
-				framework.Logf("New claim:\n%s", format.Object(claim, 1))
-				validateClaim(claim)
-			},
-			UpdateFunc: func(oldObj, newObj any) {
-				defer ginkgo.GinkgoRecover()
-				oldClaim := oldObj.(*resourceapi.ResourceClaim)
-				newClaim := newObj.(*resourceapi.ResourceClaim)
-				framework.Logf("Updated claim:\n%s\nDiff:\n%s", format.Object(newClaim, 1), cmp.Diff(oldClaim, newClaim))
-				validateClaim(newClaim)
-			},
-			DeleteFunc: func(obj any) {
-				defer ginkgo.GinkgoRecover()
-				claim := obj.(*resourceapi.ResourceClaim)
-				framework.Logf("Deleted claim:\n%s", format.Object(claim, 1))
-			},
-		})
-		framework.ExpectNoError(err, "AddEventHandler")
-		wg.Add(1)
-		go func() {
-			defer wg.Done()
-			claimInformer.Run(cancelCtx.Done())
-		}()
+		nodes.beforeEach(ctx, f, minNodes, maxNodes)
 	})
 	return nodes
 }
 
+// NewNodesNow is a variant of NewNodes which can be used inside a ginkgo.It.
+func NewNodesNow(ctx context.Context, f *framework.Framework, minNodes, maxNodes int) *Nodes {
+	nodes := &Nodes{}
+	nodes.beforeEach(ctx, f, minNodes, maxNodes)
```

Comment:
```suggestion
	nodes.newNodes(ctx, f, minNodes, maxNodes)
```
?

Any other name that could summarize the logic in here?

Reply:
How about "init"?